cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -309,6 +313,31 @@ public void testMapreduceHistory() throws Exception {
   }
 
   /**
+   * Tests history of a non existing program
+   */
+  @Test
+  public void testNonExistingProgramHistory() throws Exception {
```

Comment:
Hi @chtyim , in terms of "remove the Assert statement in the finally block" While as this assert checks if the deployed app is deleted correctly, if it fails, without this assert, it may move directly to the next test case. I wonder what is the good way to do this.